tools/azsdk-cli/Azure.Sdk.Tools.Cli/Services/Languages/GoLanguageService.cs

```diff
@@ -39,131 +35,76 @@ public GoLanguageService(
 
     public override async Task<PackageInfo> GetPackageInfo(string packagePath, CancellationToken ct = default)
     {
+        var repoRoot = gitHelper.DiscoverRepoRoot(packagePath);
+        var commonPS1 = Path.Join(repoRoot, "eng", "common", "scripts", "common.ps1");
+
+        // The powershell outputs this:
+        // {
+        //   "Name": "sdk/messaging/azservicebus",
+        //   "Version": "1.10.1-beta.1",
+        //   "DirectoryPath": "../sdk/messaging/azservicebus",
+        //   "ServiceDirectory": "messaging/azservicebus",
+        //   "ReadMePath": "../sdk/messaging/azservicebus/README.md",
+        //   "ChangeLogPath": "../sdk/messaging/azservicebus/CHANGELOG.md",
+        //   "SdkType": "client",
+        //   "IsNewSdk": true,
+        //   "ReleaseStatus": "Unreleased",
+        //   "IncludedForValidation": false,
+        //   "CIParameters": {
+        //     "CIMatrixConfigs": []
+        //   },
+        //   "VersionFile": "/home/ripark/src/az/sdk/messaging/azservicebus/internal/constants.go",
+        //   "ModuleName": "azservicebus"
+        // }
         logger.LogDebug("Resolving Go package info for path: {packagePath}", packagePath);
-        var (repoRoot, relativePath, fullPath) = Parse(gitHelper, packagePath);
-        var (packageName, packageVersion) = await TryGetPackageInfoAsync(fullPath, ct);
-        
-        if (packageName == null)
+        string[] args = ["-c", $". {commonPS1}; Get-GoModuleProperties('{packagePath}') | ConvertTo-Json"];
+        var processResult = await processHelper.Run(new ProcessOptions(
+            "pwsh", args,
+            "pwsh.exe", args,
+            workingDirectory: repoRoot), ct);
+
+        if (processResult.ExitCode != 0)
         {
-            logger.LogWarning("Could not determine package name for Go package at {fullPath}", fullPath);
+            throw new Exception($"Failed to extract package properties for {packagePath}: {processResult.Output}");
         }
-        if (packageVersion == null)
+
+        var goModuleProperties = JsonSerializer.Deserialize<GoModulePropertiesPowershell>(processResult.Output)
```

Comment:
While parsing the stdout might work in can be cause trouble if someone decides to output other random crap. We probably want to move writing this to a file instead. Similar to https://github.com/Azure/azure-sdk-for-go/blob/main/eng/common/scripts/Save-Package-Properties.ps1 but I don't think that will work in its current form, so we can stick with this for now.

Reply:
Yeah, this is always a concern - I don't know if there are good guidelines, or even enforcement, for making sure that errors/info go to stderr or not.